homework/list.js

```diff
@@ -2,78 +2,94 @@
 
 (function() {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+  var NoteListManager = function() {
+  // Local data storage; should sync up with the server.
+    this._listNoteContent = [];
+    // Will be an element as the list wrapper.
+    this._wrapper = document.querySelector('#note-list-wrapper');
+  };
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+  NoteListManager.prototype = {
+    start() {      
+      this.fetchList((function(data) {
+        var promise = Promise.resolve();
+        promise
+          .then(this.updateList(data))
+          .then(this.drawList())
+          .then(this.preloadFirstNote());
+        /*this.updateList(data);
+        this.drawList();
+        this.preloadFirstNote();*/
+      }).bind(this));
+      window.addEventListener('click', this);
+    },
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+    onNoteOpen(event) {
```

Comment:
It's a convention to prefix method with on for the corresponding function, so it should be renamed as 'onClick'. And, to set this as the listener is better than an anonymous function.